traffic-split.md

```diff
@@ -21,24 +21,76 @@ Implementations will weight outgoing traffic between the services referenced by
 `spec.backends`. Each backend is a Kubernetes service that potentially has a
 different selector and type. Weights must be whole numbers.
 
+To accommodate A/B testing scenarios, a traffic split can take into account
+HTTP header filters and route a specific user segment to a backend while
+all the other users not belonging to that segment will be routed to the
+default service backend e.g. the Kubernetes service that matches
+the *root* name. The HTTP header filters can be specified using the
+[HTTPRouteGroup](traffic-specs.md) API, a traffic split can refer HTTP route
+groups via `spec.matches` thus applying the header filters described in
+those groups.
+
 ## Specification
 
+Canary example:
+
 ```yaml
-apiVersion: split.smi-spec.io/v1alpha2
+apiVersion: split.smi-spec.io/v1alpha3
 kind: TrafficSplit
 metadata:
-  name: my-trafficsplit
+  name: canary
 spec:
   # The root service that clients use to connect to the destination application.
-  service: my-website
+  service: website
   # Services inside the namespace with their own selectors, endpoints and configuration.
   backends:
-  - service: my-website-v1
-    weight: 50
-  - service: my-website-v2
-    weight: 50
+  - service: website-v1
+    weight: 90
+  - service: website-v2
+    weight: 10
 ```
 
+The above configuration will route 10% of the `website` incoming
+traffic to the `website-v1` service and 90% to `website-v2` service.
+
+A/B test example:
+
+```yaml
+apiVersion: split.smi-spec.io/v1alpha3
+kind: TrafficSplit
+metadata:
+  name: ab-test
+spec:
+  service: website
+  matches:
```

Comment:
Should service be moved into `matches`? I think that'd clear up my confusion on this one:

```
matches:
- kind: HTTPRouteGroup
  name: ab-test
- kind: Service
  name: website
```

Reply:
This would break the current API if we move the apex to matches, also I don't think we should support multiple services like the array implies.

Reply:
Hmm, I'm okay having a deprecation cycle if it makes sense. Multiple apex services is a little weird though, I'll admit.

My primary confusion is around matching and what fits with what else. I eventually got it, just wasn't immediately intuitive.